cpp/tests/lists/contains_tests.cpp

```diff
@@ -575,6 +575,146 @@ TEST_F(ContainsTest, VectorTypeRelatedExceptions)
   }
 }
 
+template <typename T>
+struct TypedContainsNaNsTest : public ContainsTest {
+};
+
+TYPED_TEST_CASE(TypedContainsNaNsTest, FloatingPointTypes);
+
+template <typename T>
+T get_nan(const char* const nan_contents)
+{
+  return std::nan(nan_contents);
+}
+
+template <>
+float get_nan<float>(const char* const nan_contents)
+{
+  return std::nanf(nan_contents);
+}
+
+TYPED_TEST(TypedContainsNaNsTest, ListWithNaNsContainsScalar)
+{
+  using T = TypeParam;
+
+  auto nan_1 = get_nan<T>("1");
+  auto nan_2 = get_nan<T>("2");
+  auto nan_3 = get_nan<T>("3");
+
+  auto search_space = lists_column_wrapper<T>{
+    {0.0, 1.0, 2.0},
+    {3, 4, 5},
+    {6, 7, 8},
+    {9, 0, 1},
+    {nan_1, 3.0, 4.0},
+    {5, 6, 7},
+    {8, nan_2, 0},
+    {},
+    {1, 2, 3},
+    {}}.release();
+
+  auto search_key_nan = create_scalar_search_key<T>(nan_3);
+
+  auto actual_result = lists::contains(search_space->view(), *search_key_nan);
+
+  auto expected_result = fixed_width_column_wrapper<bool>{0, 0, 0, 0, 1, 0, 1, 0, 0, 0};
+
+  CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(expected_result, *actual_result);
+}
+
+TYPED_TEST(TypedContainsNaNsTest, ListWithNaNsContainsVector)
+{
+  using T = TypeParam;
+
+  auto nan_1 = get_nan<T>("1");
+  auto nan_2 = get_nan<T>("2");
+  auto nan_3 = get_nan<T>("3");
+
+  auto search_space = lists_column_wrapper<T>{
+    {0.0, 1.0, 2.0},
+    {3, 4, 5},
+    {6, 7, 8},
+    {9, 0, 1},
+    {nan_1, 3.0, 4.0},
+    {5, 6, 7},
+    {8, nan_2, 0},
+    {},
+    {1, 2, 3},
+    {}}.release();
+
+  auto search_key_values = std::vector<T>{1.0, 2.0, 3.0, nan_3, nan_3, nan_3, 0.0, nan_3, 2.0, 0.0};
+
+  auto null_at_index_2 =
```

Comment:
why this index? AFAICT it does not change the output expect for the mask. Might be good to also have a case where the input would contain the key, if the null mask was all valid.

Reply:
`null_at_index_2` was to test that searching for `∅` in a list without nulls would yield `∅`, not `false`. But I see your point: The output null-mask matches that of the `search_keys`. I'll modify this so that there are `∅`s within the input lists as well.
(This really wasn't the focus of the test, because those conditions are checked in other tests. I'll add it here for completeness.)

Reply:
Might be good to add a comment before this line to describe the intent (unless I misunderstood an obvious thing here).

Reply:
I've modified the test, and added some documentation to clarify.